src/workerd/jsg/jsg.h

```diff
@@ -783,14 +783,7 @@ class Data {
   void destroy();
 
   // Debugging helpers.
-  void assertInvariant() {
-#ifdef KJ_DEBUG
-    assertInvariantImpl();
-#endif
-  }
-#ifdef KJ_DEBUG
-  void assertInvariantImpl();
-#endif
+  void assertInvariant();
```

Comment:
This method was intentionally designed to be optimized out in release mode, since it is called every time a `Data` is moved (note that `Data` underlies `jsg::Value` and `V8Ref`).

Reply:
Ah, so we should definitely be documenting that. Is the concern that `isEmpty()` will be expensive or that throwing an exception will prevent the move ctor from being elided?

Reply:
I think just having an out-of-line function call in a move constructor likely makes life difficult for the compiler. Like I would hope the compiler can see when a variable is moved-away and then subsequently destroyed, the destructor can be elided because the move nulled out all the pointers. But if there's an out-of-line function call in the middle of the move then probably the compiler can't reason about it at all anymore.

Both performance and code bloat are concerns since we pass V8Refs by move in quite a few places.

Of course, if actual measurements show this has negligible impact, then I don't have a problem. But I dunno if it's worth spending time to determine that, vs. just leaving this as it is. I feel like the check here should provide sufficient coverage running only in debug mode, and I'm not even sure if I've ever seen it fail in any case.